app/jobs/check_catalogues_job.rb

```diff
@@ -0,0 +1,89 @@
+class CheckCataloguesJob < ActiveJob::Base
+  queue_as :default
+
+  require 'capybara'
+  require 'capybara/poltergeist'
+  require_relative '../../app/jobs/scrape_catalogues_job.rb'
+  require_relative '../../app/jobs/delete_catalogues_job.rb'
+  require_relative '../../app/jobs/capybara_quit_job.rb'
+
+  include Capybara::DSL
+  Capybara.register_driver :poltergeist do |app|
+    Capybara::Poltergeist::Driver.new(app, timeout: 60, js_errors: false)
+  end
+  Capybara.default_driver = :poltergeist
+
+  def perform
+
+    #new_catalogues stores result from calling catalogue_gather method
+    new_catalogues = catalogue_gather
+
+    #collect current_catalogue data
+    current_catalogue = Hash.new(0)
+    Catalogue.all.each{|c| current_catalogue[c.id] = c.catalogue_num}
+
+    #scrape new posts from new catalogue
+    catalogue_to_add = add_new_catalogue(current_catalogue, new_catalogues)
+    ScrapeCataloguesJob.perform_later(catalogue_to_add)
+
+    #delete old and duplicate catalogues
+    delete_old_duplicate_catalogue(current_catalogue, new_catalogues, catalogue_to_add)  

```

Comment:
I tried this and ran into error ```ActiveRecord::StatementInvalid: SQLite3::SQLException: near ",": syntax error: SELECT "catalogues".* FROM "catalogues" WHERE (catalogue_num not in '8327','8271','8317','8282')
```

in this example, the published_catalogue_nums = ['8327', '8271', '8317', '8282'], and I have existing catalogue = '1234'.

I'm not familiar with code that directly deal with SQL. perhaps could need your help with this.

I really like how you've refactor this method to just few lines. really curious to get this code working.